src/main/java/de/tum/cit/aet/artemis/lti/web/LtiResource.java

```diff
@@ -121,32 +123,62 @@ public ResponseEntity<OnlineCourseConfiguration> updateOnlineCourseConfiguration
+     * @param competency           A flag indicating whether the deep linking request is for a competency resource.
+     * @param learningPath         A flag indicating whether the deep linking request is for a learning path resource.
+     * @param iris                 A flag indicating whether the deep linking request is for an IRIS integration.
+     * @param ltiIdToken           The token containing the deep linking information.
+     * @param clientRegistrationId The identifier of the course's online configuration.
      * @return A ResponseEntity containing a JSON object with the 'targetLinkUri' property set to the deep linking response target link.
+     * @throws ParseException           If the LTI ID token cannot be parsed.
+     * @throws BadRequestAlertException If LTI is not configured for the course or if no valid deep linking type is provided.
      */
     @PostMapping("lti13/deep-linking/{courseId}")
-    @EnforceAtLeastInstructor
-    public ResponseEntity<String> lti13DeepLinking(@PathVariable Long courseId, @RequestParam(name = "exerciseIds") Set<Long> exerciseIds,
+    @EnforceAtLeastInstructorInCourse
+    public ResponseEntity<String> lti13DeepLinking(@PathVariable Long courseId, @RequestParam(name = "exerciseIds", required = false) Set<Long> exerciseIds,
+            @RequestParam(name = "lectureIds", required = false) Set<Long> lectureIds, @RequestParam(name = "competency", required = false) boolean competency,
+            @RequestParam(name = "learningPath", required = false) boolean learningPath, @RequestParam(name = "iris", required = false) boolean iris,
             @RequestParam(name = "ltiIdToken") String ltiIdToken, @RequestParam(name = "clientRegistrationId") String clientRegistrationId) throws ParseException {
-        log.info("LTI 1.3 Deep Linking request received for course {} with exercises {} for registrationId {}", courseId, exerciseIds, clientRegistrationId);
+
+        log.info("LTI 1.3 Deep Linking request received for course {} with exercises: {}, lectures: {}, competency: {}, learningPath: {}, iris: {}, registrationId: {}", courseId,
+                exerciseIds, lectureIds, competency, learningPath, iris, clientRegistrationId);
 
         Course course = courseRepository.findByIdWithEagerOnlineCourseConfigurationElseThrow(courseId);
-        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
 
         if (!course.isOnlineCourse() || course.getOnlineCourseConfiguration() == null) {
             throw new BadRequestAlertException("LTI is not configured for this course", "LTI", "ltiNotConfigured");
         }
 
         OidcIdToken idToken = new OidcIdToken(ltiIdToken, null, null, SignedJWT.parse(ltiIdToken).getJWTClaimsSet().getClaims());
 
-        String targetLink = ltiDeepLinkingService.performDeepLinking(idToken, clientRegistrationId, courseId, exerciseIds);
+        String targetLink;
+
+        if (exerciseIds != null) {
```

Comment:
The long if-else block looks ugly. I wonder if there are other solutions to solve this.